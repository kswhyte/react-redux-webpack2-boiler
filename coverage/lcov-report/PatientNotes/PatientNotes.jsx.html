<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for PatientNotes/PatientNotes.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">PatientNotes</a> PatientNotes.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import Moment from 'moment';
import './patient-notes.css';
&nbsp;
const PatientNotes = props =&gt; {
  PatientNotes.propTypes = {
    children: PropTypes.object,
    notes: PropTypes.array.required
  };
&nbsp;
  const check = () =&gt; {
    console.log(props);
  };
  check();
&nbsp;
  return (
    &lt;div className="patient-notes-wrapper"&gt;
      &lt;div className="tab-content clearfix"&gt;
        &lt;div className="patient-notes-form tab-pane active"&gt;
          {
            (props.notes.length === 0) &amp;&amp; (
              &lt;div className="page-instructions"&gt;
                &lt;p&gt;No notes description&lt;/p&gt;
              &lt;/div&gt;
            )
          }
          &lt;table className="notes-tables header-table"&gt;
            &lt;thead&gt;
              &lt;th&gt;Date&lt;/th&gt;
              &lt;th&gt;Time&lt;/th&gt;
              &lt;th&gt;Title&lt;/th&gt;
              &lt;th&gt;Actions&lt;/th&gt;
            &lt;/thead&gt;
      &lt;tbody&gt;
      {
        props.notes.map(<span class="fstat-no" title="function not covered" >no</span>te =&gt; {
<span class="cstat-no" title="statement not covered" >          return (</span>
            &lt;tr key={note.noteId} &gt;
              &lt;td&gt;{Moment(note.dateTime).format('LL')}&lt;/td&gt;
              &lt;td&gt;{Moment(note.dateTime).format('LT')}&lt;/td&gt;
              &lt;td&gt;{note.description}&lt;/td&gt;
              &lt;td&gt;&lt;/td&gt;
            &lt;/tr&gt;
          )
        })
      }
      &lt;/tbody&gt;
    &lt;/table&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
&nbsp;
&nbsp;
PatientNotes.defaultProps = {
  notes: []
};
&nbsp;
//TODO: make sure data is coming in for each note. UNCOMMENT when Redux wiring is setup
// const ExistingNotes = props =&gt; {
//   ExistingNotes.propTypes = {
//     existingNotes: PropTypes.array.required
//   };
//   return (
//     &lt;table className="notes-tables"&gt;
//       {props.existingNotes.map(note =&gt; {
//         return (
//           &lt;tr key={note.noteID}&gt;
//             &lt;td&gt;
//               {Moment(note.noteDateTime).format('LL')}
//             &lt;/td&gt;
//             &lt;td&gt;
//               {Moment(note.noteTitle).format('LT')}
//             &lt;/td&gt;
//           &lt;/tr&gt;
//         );
//       })}
//     &lt;/table&gt;
//   );
// };
&nbsp;
export default PatientNotes;
&nbsp;
&nbsp;
&nbsp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Aug 09 2017 12:54:54 GMT-0600 (MDT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
