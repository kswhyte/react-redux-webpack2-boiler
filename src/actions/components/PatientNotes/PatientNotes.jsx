import React from 'react';
import PropTypes from 'prop-types';

const PatientNotes = props => {
  PatientNotes.propTypes = {
    children: PropTypes.object
  };

  const check = () => {
    console.log(props);
  };
  check();

  return (
    <div className="patient-notes-wrapper">
      <div className="tab-content clearfix">
        <div className="patient-notes-form tab-pane active">
          <h3>Insert NOTES Form</h3>
        </div>
      </div>
    </div>
  );
};

export default PatientNotes;
