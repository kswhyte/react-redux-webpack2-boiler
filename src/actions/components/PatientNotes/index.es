import Component from './PatientNotes';
export default Component;
