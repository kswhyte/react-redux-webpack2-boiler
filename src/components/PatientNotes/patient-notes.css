@import "imports/_helpers.css";

.patient-notes-wrapper {

    & .page-instructions {
        margin-top: 20px;
        width: 100%;
        text-align: center;
    }

    & .notes-tables {
        margin-top: 20px;
        width: 100%;
    }
    & .notes-tables:last-child {
        margin-bottom: 40px;
    }
    & .header-table {
        margin-top: 20px;
    }
    & .header-table th {
        border-bottom: solid 3px var(--light-grey);
    }
    & .notes-tables th {
        padding: 0px 20px 5px;
        font-family: Open Sans;
        font-size: 12px;
        font-weight: bold;
    }


    & .notes-tables td {
        width: 16.666%;
    }
    & .notes-tables td {
        font-family: Open Sans;
        font-size: 12px;
        padding: 20px 10px 20px;
        empty-cells: show;
    }
    & .notes-tables tr {
        border-bottom: solid 2px var(--light-grey);
    }

    & td > button {
        background: var(--white);
        color: var(--primary-color);
        border: 1px solid var(--primary-color);
        position: relative;
        display: inline-block;
        box-sizing: border-box;
        padding: 6px 20px;
        font-family: 'Open Sans';
        font-size: 10px;
        line-height: 1;
        text-decoration: none;
        text-align: center;
        cursor: pointer;
        border-radius: 0px;
    }

}

