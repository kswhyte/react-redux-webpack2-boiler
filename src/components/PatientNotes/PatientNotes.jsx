import React from 'react';
import PropTypes from 'prop-types';
import Moment from 'moment';
import './patient-notes.css';

const PatientNotes = props => {
  PatientNotes.propTypes = {
    children: PropTypes.object,
    notes: PropTypes.array.required
  };

  const check = () => {
    console.log(props);
  };
  check();

  return (
    <div className="patient-notes-wrapper">
      <div className="tab-content clearfix">
        <div className="patient-notes-form tab-pane active">
          {
            (props.notes.length === 0) && (
              <div className="page-instructions">
                <p>No notes description</p>
              </div>
            )
          }
          <table className="notes-tables header-table">
            <thead>
              <th>Date</th>
              <th>Time</th>
              <th>Title</th>
              <th>Actions</th>
            </thead>
      <tbody>
      {
        props.notes.map(note => {
          return (
            <tr key={note.noteId} >
              <td>{Moment(note.dateTime).format('LL')}</td>
              <td>{Moment(note.dateTime).format('LT')}</td>
              <td>{note.description}</td>
              <td></td>
            </tr>
          )
        })
      }
      </tbody>
    </table>
        </div>
      </div>
    </div>
  );
};


PatientNotes.defaultProps = {
  notes: []
};

//TODO: make sure data is coming in for each note. UNCOMMENT when Redux wiring is setup
// const ExistingNotes = props => {
//   ExistingNotes.propTypes = {
//     existingNotes: PropTypes.array.required
//   };
//   return (
//     <table className="notes-tables">
//       {props.existingNotes.map(note => {
//         return (
//           <tr key={note.noteID}>
//             <td>
//               {Moment(note.noteDateTime).format('LL')}
//             </td>
//             <td>
//               {Moment(note.noteTitle).format('LT')}
//             </td>
//           </tr>
//         );
//       })}
//     </table>
//   );
// };

export default PatientNotes;



