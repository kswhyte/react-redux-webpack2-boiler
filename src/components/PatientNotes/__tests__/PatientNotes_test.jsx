import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import PatientNotes from '../PatientNotes';

// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { shallow } from 'enzyme';
import expect from 'expect';

const stories = storiesOf('PatientNotes', module);

stories.add('Story', () => {
  const patientNotesStory = <PatientNotes />;

  specs(() =>
    describe('Show a successful alert', () => {
      it('Should render the PatientNotes component without crashing', () => {
        let output = shallow(<PatientNotes />);
        //act
        const lengthOf = output.find('.patient-notes-wrapper');
        expect(lengthOf.length).toEqual(1)
      });
    })
  );
  return patientNotesStory;
});
