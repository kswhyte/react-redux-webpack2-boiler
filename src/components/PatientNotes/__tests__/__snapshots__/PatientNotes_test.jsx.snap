// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story 1`] = `
<div
  className="patient-notes-wrapper"
>
  <div
    className="tab-content clearfix"
  >
    <div
      className="patient-notes-form tab-pane active"
    >
      <div
        className="page-instructions"
      >
        <p>
          No notes description
        </p>
      </div>
      <table
        className="notes-tables header-table"
      >
        <tr>
          <th>
            Date
          </th>
          <th>
            Time
          </th>
          <th>
            Title
          </th>
          <th>
            Actions
          </th>
        </tr>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
